util/chplenv/overrides.py

```diff
@@ -181,42 +181,45 @@ def parse(self):
                 # Strip comments and trailing white space from line
                 line = line.split('#')[0].strip()
 
-                if self.skip_line(line, linenum):
+                res = self.check_line(line, linenum)
+                if not res:
                     continue
 
-                var, val = [f.strip() for f in line.split('=', maxsplit=1)]
+                var, val = res
                 self.chplconfig[var] = val
 
-    def skip_line(self, line, linenum):
+    def check_line(self, line, linenum):
         """
         Check the various conditions for skipping a line, accumulate warnings.
+
+        Returns the tuple (var, val) if the line is valid, otherwise None.
         """
 
         # Check if line is comment, by taking length of stripped line
         if len(line) == 0:
-            return True
+            return None
 
-        # Check for syntax errors
-        try:
-            var, val = [f.strip() for f in line.split('=')]
-        except ValueError:
+
+        parts = line.split('=', maxsplit=1)
+        var = parts[0].strip()
+        val = parts[-1].strip()
```

Comment:
I'm thinking about how you can silence the warning if there's a leading equal sign in a variable for real, and I'm coming up with nothing. I don't think you should strip the string for the purposes of warning about it. So:

```
A==B # warns
A == B # warns
A = =B # doesn't warn
```

Reply:
This was an intentional choice that `A = = B` still warns. I think its more likely to be a mistake than truly what a user wants. If they want a value to start with `=`, writing `A = '= B'` should work

Reply:
As long as there's a way to silence the warning (A='=B' seems like it is one), I'm happy.

Reply:
... Mind adding a test for `A = '= B'?`